### PR TITLE
Switch to path routing (from hash routing)

### LINKED ISSUE #1712: Sources page takes you to the wrong page
As a user, when I go to the source detail page and click a Kustomization or Helm Release I am taken back to the source page.  The bug seems to be that it appends the URL instead of replacing it:

`http://localhost:8082/git_repo?name=flux-system&namespace=flux-system#/kustomization?name=flux-system&namespace=flux-system`

Expectation, when I click the `kustomization` or `helm releases` name I am taken to the correct detail page.

Click `flux-system` on table:

### PR DESCRIPTION
**Pull request by jpellizzari**
Closes #1712 

Hash routing was a bad idea. Sub-path routing is the way to go. This should work for EE as well.

### CODE REVIEW

File: ui/components/KustomizationDetail.tsx [hunk 1]
```diff
@@ -54,27 +54,27 @@ export default function KustomizationDetail({ kustomization, name }: Props) {
           suspended={kustomization?.suspended}
         />
       </Flex>
-      <TabContent wide column align>
```
Comment (joshri) [inline ui/components/KustomizationDetail.tsx:57]: I THINK this is what's causing this on my big screen 
image
Pretty much every container for the tab stuff needs `wide` and `tall`

File: ui/components/SubRouterTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    throw new Error("HashRouterTabs children must be of type HashRouterTab");
+  }
+
+  const routes: PathConfig[] = _.map(childs, (c: any) => ({
+    path: c?.props?.path,
+    name: c?.props?.name,
+  }));
+
+  return (
+    <Flex wide column start className={className}>
```
Comment (joshri) [inline ui/components/SubRouterTabs.tsx:64]: might need `tall` also

**Review by joshri — COMMENTED** [review]
Looks good - the spacing got a little weird but it should an easy fix

File: ui/components/KustomizationDetail.tsx (hunk 1 above)
Comment (jpellizzari) [reply ui/components/KustomizationDetail.tsx:57]: Cannot reproduce:
Screenshot from 2022-04-12 08-27-56

I don't think the `tall` attribute exists on my branch?

Comment (jpellizzari) [reply ui/components/KustomizationDetail.tsx]: Once I rebased, I saw this. Fixed.

**Comment by AlinaGoaga** [comment]
Thanks Jordan! Do you have a remote cluster that you use for testing with some resources on it?

**Review by joshri — APPROVED** [review]

**Comment by jpellizzari** [comment]
> Thanks Jordan! Do you have a remote cluster that you use for testing with some resources on it?

@AlinaGoaga I do not. Someone is working on that I believe :smile:  (cc @mnowster @SamLR)

**Pull request merged**